### PR TITLE
Configure session with talisker

### PR DESCRIPTION
**Pull request by tbille**
## Done

- configure session to get prometheus infor from talisker configuration.

## QA

- Check out this feature branch
- create file `.env.local` with this line: `TALISKER_NETWORKS=172.17.0.1`
- Run the site using the command `./run serve`
- View the site locally in your web browser at: [http://0.0.0.0:8023/](http://0.0.0.0:8023/)
- go on http://0.0.0.0:8023/_status/metrics
- Make sure you have the new metrics: `wsgi_latency_bucket`, `requests_latency_bucket`

## Issue / Card

Fixes https://github.com/ubuntudesign/base-squad/issues/370

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -23,6 +24,9 @@
 app.url_map.converters["regex"] = helpers.RegexConverter
 talisker.flask.register(app)
 
+if not app.testing:
```
Comment (nottrobin) [inline app.py]: Why if not testing? Wouldn't we want to test with the same requests library as we use in production?

File: app.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 # Third-party
 import flask
 import talisker.flask
+import talisker.requests
```
Comment (nottrobin) [inline app.py]: Why not just `import talisker`?

**Comment by nottrobin** [comment]
@tbille there's a simpler QA step you could have provided, simply:

``` bash
./run --env TALISKER_NETWORKS=172.17.0.1
```

File: app.py (hunk 2 above)
Comment (tbille) [reply app.py]: I don't know how talisker imports are configured but it didn't work for me

**Comment by tbille** [comment]
@nottrobin true, I added the file for me so that I don't have to care about it.

**Comment by nottrobin** [comment]
When QAing. here's my metrics: https://gist.github.com/nottrobin/98bc0740143951ad6a51b7aff6427d2f

First, we can see that the old `feed_request_latency_seconds` metrics are still there - I assume this work is intended to replace them:

```
feed_request_latency_seconds_sum{code="200",domain="admin.insights.ubuntu.com"} 10.378936
feed_request_latency_seconds_bucket{code="200",domain="admin.insights.ubuntu.com",le="0.25"} 10.0
```

Should these metrics be removed from `feeds.py`?

I'm a bit confused that the number of requests tracked by talisker seems to be so much higher than by the existing prometheus configuration, though I can believe Talisker has it right.

But also it says `view=unknown`. Should this contain something? If not, it's probably still an improvement.

```
requests_latency_bucket{host="admin-insights-ubuntu-com",le="64.0",status="200",view="unknown"} 39.0
requests_latency_bucket{host="admin-insights-ubuntu-com",le="128.0",status="200",view="unknown"} 41.0
```

**Comment by tbille** [comment]
@nottrobin 

> But also it says `view=unknown`. Should this contain something? If not, it's probably still an improvement.

this comes from the response when doing a request. On snapcraft.io we get some really usefull information, like the endpoint hitted. But this is sent in the headers which wordpress doesn't

> Should these metrics be removed from `feeds.py`?

Yes I left them to see what were the main differences, some kind of transition. But we can remove them.

**Comment by nottrobin** [comment]
> Yes I left them to see what were the main differences, some kind of transition. But we can remove them.

We don't currently consume the old metrics do we? 'Cos if not, I think we might as well remove them now, because we'll probably forget to do it later.

**Comment by tbille** [comment]
@nottrobin old metrics removed since unused

**Review by nottrobin — APPROVED** [review]
:+1: LGTM

**Pull request merged**